raiden/network/proxies/user_deposit.py

```diff
@@ -577,3 +658,161 @@ def _deposit_check_result(
+
+        if current_balance < amount_to_plan_withdraw:
+            raise BrokenPreconditionError(
+                f"Can't create WithdrawPlan for amount {amount_to_plan_withdraw}, "
+                f"it exceeds the current balance of {current_balance}."
+            )
+
+    def _withdraw_preconditions(
+        self, amount_to_withdraw: TokenAmount, given_block_identifier: BlockIdentifier
+    ) -> None:
+        try:
+            withdraw_plan = self.get_withdraw_plan(
+                withdrawer_address=self.node_address, block_identifier=given_block_identifier
+            )
+            whole_balance = self.whole_balance(block_identifier=given_block_identifier)
+        except ValueError:
+            # If 'given_block_identifier' has been pruned, we can't perform the check
+            return
+        except BadFunctionCallOutput:
+            raise_on_call_returned_empty(given_block_identifier)
+
+        if amount_to_withdraw > withdraw_plan.withdraw_amount:
+            raise BrokenPreconditionError(
+                f"Can't withdraw {amount_to_withdraw}, "
+                f"current withdraw plan only allows for {withdraw_plan.withdraw_amount}."
+            )
+
+        given_block_number = self.client.get_block(given_block_identifier)["number"]
+
+        if withdraw_plan.withdraw_block > given_block_number:
+            raise BrokenPreconditionError(
+                f"Can't withdraw at block {given_block_number}. "
+                f"The current withdraw plan requires block number {withdraw_plan.withdraw_block}."
+            )
+
+        if whole_balance - amount_to_withdraw < 0:
+            raise BrokenPreconditionError(
+                f"The current whole balance is {whole_balance}. "
+                f"The withdraw of {amount_to_withdraw} would lead to an underflow."
+            )
+
+    def _plan_withdraw_check_result(
+        self, transaction_sent: Optional[TransactionSent], amount_to_plan_withdraw: TokenAmount
+    ) -> Optional[TransactionMined]:
+        if transaction_sent is None:
+            failed_at = self.client.get_block(BLOCK_ID_LATEST)
+            failed_at_blocknumber = failed_at["number"]
+
+            self.client.check_for_insufficient_eth(
+                transaction_name="planWithdraw",
+                transaction_executed=False,
+                required_gas=self.gas_measurements["UserDeposit.planWithdraw"],
+                block_identifier=failed_at_blocknumber,
+            )
+            raise RaidenRecoverableError(
+                "Plan withdraw transaction failed to be sent for an unknown reason."
+            )
+
+        transaction_mined = self.client.poll_transaction(transaction_sent)
+
+        if not was_transaction_successfully_mined(transaction_mined):
+            if amount_to_plan_withdraw <= 0:
+                raise RaidenRecoverableError(
+                    f"Planned withdraw amount was <= 0: {amount_to_plan_withdraw}."
```

Comment:
This should never happen, `amount_to_plan_withdraw` is a value object that does not change with the blockchain and thus does not depend on the block being available or its number. I'd say this is not necessary.

Reply:
It can happen if the preconditions are skipped because of pruned blocks and nothing else on higher layers prevents a value of `<= 0` being passed in.

As I understood the other proxies I looked at as inspiration all those post-checks are simply there to provide better error messages.

Reply:
Yes, all these checks are just for better error messages since debugging these errors manually takes a lot of work. Even with these messages it is time consuming to understand what went wrong.

The point here is that `amount_to_plan_withdraw <= 0` should not be skipped in the precondition check if the block is pruned, because this does not depend on the blockchain data. But this is a minor detail we don't even need to change now